### PR TITLE
Add theme support

### PR DESCRIPTION
**Pull request by obedparla**
- Create a theme (copied from https://github.com/heetch/flamingo/pull/136)
- Replace all Flamingo vars with the theme. 
- Replace media queries for the theme media queries 🎉 
- Export the theme so external products can use it instead of the flamingo vars

### The file changes are massive, but it's basically a find-replace-all of the vars for the theme counterpart. I double-checked every single file and tested all components again to make sure everything still worked 😄

### CODE REVIEW

File: src/theme.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+export const theme = {
+  borderRadius: {
+    xl: 'var(--f-borderRadius--xl)',
+    l: 'var(--f-borderRadius--l)',
+    m: 'var(--f-borderRadius--m)',
+    s: 'var(--f-borderRadius--s)',
+  },
+  breakPoint: {
+    sUp: '@media (min-width: 460px)',
```
Comment (ugogo) [inline src/theme.js:9]: `sUp`? 🤔

File: src/components/FileUploader/index.js [hunk 2]
```diff
@@ -181,14 +176,9 @@ FileUploader.propTypes = {
 
 FileUploader.defaultProps = {
   translate: ({ defaultText }) => defaultText,
-  hasError: false,
```
Comment (ugogo) [inline src/components/FileUploader/index.js:184]: Why did you remove all those props?

File: src/theme.js (hunk 1 above)
Comment (obedparla) [reply src/theme.js:9]: a bit of an awkward name 😄 . It's for `s` but up. Any better name comes to mind?

File: src/components/FileUploader/index.js (hunk 2 above)
Comment (obedparla) [reply src/components/FileUploader/index.js:184]: `false` and `undefined` are both falsy, so they don't provide any value by being false, just extra code IMO. I checked that it doesn't change functionality

Comment (ugogo) [reply src/components/FileUploader/index.js:184]: 👌

File: src/theme.js (hunk 1 above)
Comment (ugogo) [reply src/theme.js:9]: Oh okay. Not really haha
